packages/dataviews/src/search-widget.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+import { matchSorter } from 'match-sorter';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo, useDeferredValue } from '@wordpress/element';
+import {
+	BaseControl,
+	Icon,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { check, search } from '@wordpress/icons';
+
+export default function SearchWidget( { filter, view, onChangeView } ) {
+	const [ searchValue, setSearchValue ] = useState( '' );
+	const deferredSearchValue = useDeferredValue( searchValue );
+	const selectedFilter = view.filters.find(
+		( _filter ) => _filter.field === filter.field
+	);
+	const selectedValues = selectedFilter?.value;
+	const matches = useMemo( () => {
+		return matchSorter( filter.elements, deferredSearchValue ?? '', {
+			keys: [ 'label' ],
+		} );
+	}, [ filter.elements, deferredSearchValue ] );
+	return (
+		<Ariakit.ComboboxProvider
+			value={ searchValue }
+			setSelectedValue={ ( value ) => {
+				const currentFilter = view.filters.find(
+					( _filter ) => _filter.field === filter.field
+				);
+				const newFilters = currentFilter
+					? [
+							...view.filters.map( ( _filter ) => {
+								if ( _filter.field === filter.field ) {
+									return {
+										..._filter,
+										operator:
+											currentFilter.operator ||
+											filter.operators[ 0 ],
+										value,
+									};
+								}
+								return _filter;
+							} ),
+					  ]
+					: [
+							...view.filters,
+							{
+								field: filter.field,
+								operator: filter.operators[ 0 ],
+								value,
+							},
+					  ];
+				onChangeView( {
+					...view,
+					page: 1,
+					filters: newFilters,
+				} );
+			} }
+			setValue={ setSearchValue }
+		>
+			<BaseControl
+				className="dataviews-search-widget-filter-combobox__wrapper"
+				__nextHasNoMarginBottom
+			>
+				<div className="dataviews-search-widget-filter-combobox__input-wrapper">
+					<Ariakit.Combobox
+						autoSelect
```

Comment:
I tried to pass `SearchControl` to `render` prop here, but there is an issue because Ariakit expects always an event at the `onChange` callback([destructures](https://github.com/ariakit/ariakit/blob/main/packages/ariakit-react-core/src/combobox/combobox.ts#L364) from `event.target`), so we can't use in combination with `setValueOnChangeProp` prop set to `false`.

For this reason I copied some html structure and styles from `SearchControl`, but also placed the icon at the left side.

Reply:
cc @mirka who has been working on refreshing `SearchControl` lately

Reply:
Maybe there's a way for us to hack around that — ie. something like this?

```
const HackedSearchControl = ( { onChange, ...props } ) => {
	const [ inputRef, setInputRef ] = useState( null );

	useEffect( () => {
		inputRef?.addEventListener( 'change', onChange );

		return () => {
			inputRef?.removeEventListener( 'change', onChange );
		};
	}, [ inputRef, onChange ] );

	return (
		<SearchControl { ...props } ref={ ( node ) => setInputRef( node ) } />
	);
};


// ... later


<Ariakit.Combobox
	autoSelect
	placeholder={ __( 'Search' ) }
	className="dataviews-search-widget-filter-combobox__input"
	render={ <HackedSearchControl /> }
/>
```

Haven't tried it on my machine, though — but it could be an idea